ert3/stats/_stats.py

```diff
@@ -47,8 +47,8 @@ def _to_record(self, x: np.ndarray) -> ert.data.Record:  # type: ignore
             return ert.data.NumericalRecord(data=x.tolist())
         else:
             return ert.data.NumericalRecord(
-                data={
-                    idx: float(val) for idx, val in zip(self.index, x)  # type: ignore
+                data={  # type: ignore
```

Comment:
Why's the difference here? Another black version?

Reply:
mypy was complaining on github workflow typing. speculating here but I think it had to do with yesterday's bumped version, which has already been fixed. will try it again with mypy old version.

Reply:
weird, the problem still persists, so how did it pass when it first got written?

Reply:
```bash
❯ mypy --version
mypy 0.910
❯ mypy -p ert -p ert3
ert/data/record/_record.py:22: error: Module "beartype.roar" has no attribute "BeartypeDecorHintPepDeprecationWarning"; maybe "BeartypeDecorHintPepDeprecatedWarning", "BeartypeDecorHintPepWarning", or "BeartypeDecorHintPepException"?
ert3/stats/_stats.py:50: error: Argument "data" to "NumericalRecord" has incompatible type "Dict[object, float]"; expected "Union[List[Union[int, float]], Dict[str, Union[int, float]], Dict[int, Union[int, float]]]"
Found 2 errors in 2 files (checked 42 source files)
```

Reply:
Right, the thing self.index can be of two types in `NumericalRecord`; ie. `RecordIndex = Union[Tuple[int, ...], Tuple[str, ...]]` 
The best approach would be to check `isinstance(self.index[0], str)` and `isinstance(self.index[0], int)` I reckon ...

Reply:
I also understand that this is not part of your PR. Not sure what's the best approach.

Reply:
this issue is being handled here: #2642